src/ontodia/diagram/paperArea.tsx

```diff
@@ -674,6 +723,124 @@ export class PaperArea extends React.Component<PaperAreaProps, State> {
+        }
+
+        body();
+
+        const duration = typeof durationMs === 'number' ? durationMs : DEFAULT_ANIMATION_DURATION;
+        return new Promise(resolve => setTimeout(() => {
+            resolve();
+            this.stopCssAnimation();
+        }, duration));
+    }
+
+    private stopCssAnimation = () => {
+        if (this.cssAnimations === 0) { return; }
+
+        const previous = this.animated;
+        this.cssAnimations--;
+        const value = this.animated;
+
+        if (previous !== value) {
+            this.forceUpdate();
+            this.source.trigger('changeAnimated', {source: this, previous: previous});
+        }
+    }
+
+    private get viewportState(): ViewportState {
+        const {clientWidth, clientHeight} = this.area;
+        const {originX, originY, paddingX, paddingY, scale} = this.state;
+
+        const scrollCenterX = this.area.scrollLeft + clientWidth / 2 - paddingX;
+        const scrollCenterY = this.area.scrollTop + clientHeight / 2 - paddingY;
+        const paperCenter = {
+            x: scrollCenterX / scale - originX,
+            y: scrollCenterY / scale - originY,
+        };
+
+        return {
+            center: paperCenter,
+            scale: {
+                x: scale,
+                y: scale,
+            }
+        };
+    }
+
+    private setViewportState(state: Partial<ViewportState>, animate?: ViewportAnimationOptions): Promise<void> {
+        if (this.viewportAnimation) {
+            this.viewportAnimation.cancellation.abort();
+        }
+        const from = this.viewportState;
+        const to = {
+            ...from,
+            ...state,
+        };
+        if (animate) {
+            const viewportAnimation: ViewportAnimation = {
+                from, to, cancellation: new Cancellation(),
+            };
+            const animationDescription = animate instanceof Object ?
+                animate as ViewPortAnimationDescription : undefined;
+            const durationMs = animationDescription && animationDescription.duration ?
+                animationDescription.duration : DEFAULT_ANIMATION_DURATION;
+
+            const awaitPromise = animateInterval(progress => {
+                const curState: ViewportState = {
+                    center: {
+                        x: from.center.x + (to.center.x - from.center.x) * progress,
+                        y: from.center.y + (to.center.y - from.center.y) * progress,
+                    },
+                    scale: {
+                        x: from.scale.x + (to.scale.x - from.scale.x) * progress,
+                        y: from.scale.y + (to.scale.y - from.scale.y) * progress,
+                    }
+                };
+                this.applyViewportState(curState);
+            }, durationMs, viewportAnimation.cancellation);
+
+            this.viewportAnimation = viewportAnimation;
+            return awaitPromise.then(() => {
+                delete this.viewportAnimation;
```

Comment:
It's better to assign to `undefined` here instead of using `delete` operator:
```ts
this.viewportAnimation = undefined;
```
> Does using delete keyword affect v8 optimizations of an object? TL;DR as a general rule of thumb, using 'delete' makes thing slower.
See https://stackoverflow.com/questions/14967535/delete-a-x-vs-a-x-undefined#comment59269653_29600852